subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginExecutionPhaseStartedCallbackIntegrationTest.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class GradleEnterprisePluginExecutionPhaseStartedCallbackIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
should we add a test about a build tree with actually several builds, to make sure it's really called only once per build **tree** ?

Reply:
Will do that, thanks.

Reply:
Also probably worth covering the included build for setting plugin case as there are some behavioural differences there.

Reply:
@mlopatkin see for instance the reproducer project that Jendrik linked to on https://github.com/gradle/gradle/issues/19077
It contains two settings plugin created by included build cases, class-based or with a precompiled script plugin

Reply:
I applied precompiled and class-based plugins to settings and projects to cover most of the code paths. I don't think we need separate tests for each case, don't we?